### DISCUSSION TITLE
How to setup host and port manually?

### DISCUSSION THREAD

**Discussion by Klassikcat** [Lightning-AI/pytorch-lightning#17316] [Lightning App API: LightningApp, LightningFlow, LightningWork]
Hello, i'm using lightningapp to serve my model. it automatically uses localhost:7501, but i need to set it to 0.0.0.0:7501 to send request from outside of my private network.

Is there any way to set host and port maunally in lightningapp or lightningflow?

**Comment by Yosshi999** [comment]
Set environment variable `LIGHTNING_APP_STATE_URL=http://0.0.0.0`
See https://github.com/Lightning-AI/lightning/blob/deb70464967f4a996714874ebf44861c3481e0ff/src/lightning/app/core/constants.py#L39-L41

But you may need to open an additional port and modify codes in a subclass of `LightningWork`.
To use Gradio in Lightning, you have to modify [launch options](https://github.com/Lightning-AI/lightning/blob/deb70464967f4a996714874ebf44861c3481e0ff/src/lightning/app/components/serve/gradio_server.py#L97-L98).
